controllers/product.go

```diff
@@ -0,0 +1,178 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/cloudinary/cloudinary-go/api/uploader"
+	"github.com/distributed-marketplace-system/db"
+	"github.com/distributed-marketplace-system/models"
+	"github.com/distributed-marketplace-system/util"
+	"github.com/gin-gonic/gin"
+)
+
+//ProductController ...
+type ProductController struct{}
+
+func UploadProductImage(img_path string) (url string, err error) {
+
+	if _, err := os.Stat(img_path); os.IsNotExist(err) {
+		return "", err
+	}
+
+	var ctx = context.Background()
+	uploadResult, err := util.CLD.Upload.Upload(ctx, img_path, uploader.UploadParams{Folder: "asu"})
+	if err != nil {
+		log.Fatalf("Failed to upload file, %v\n", err)
+		return "", err
+	}
+
+	log.Println(uploadResult.SecureURL)
+	return uploadResult.SecureURL, nil
+}
+
+func (ctrl ProductController) AddProduct(c *gin.Context) {
+
+	// Get the email of the user to compare with email in the header
```

Comment:
Actually, we shouldn't ask the user for his Id, we should already know this info from the token, we can add the userId to the token if this will be easier for us